examples/contrib/readhar.py

```diff
@@ -0,0 +1,141 @@
+        """
+        Creates a HTTPFlow object from a given entry in HAR file
+        """
+
+        request_method = request_json["request"]["method"]
+        request_url = request_json["request"].get("url", None)
+        if not request_url:
+            request_url = request_json["request"].get("host", None)
+
+        request_headers = self.fix_headers(request_json["request"].get("headers", []))
+
+        client_conn = connection.Client(
+            peername=("127.0.0.1", 51513),
+            sockname=("127.0.0.1", 8080),
+            # TODO Get time info from HAR File
+            timestamp_start=time.time(),
+        )
+
+        # 375:3 is default mitmproxy server_conn when making a new flow.
+        server_conn = connection.Server(address=("375", 3))
+
+        newflow = http.HTTPFlow(client_conn, server_conn)
+        try:
+            newflow.request = http.Request.make(
+                request_method, request_url, "", request_headers
+            )
+        except TypeError as e:
+            logger.error("Failed to create request")
+            raise exceptions.OptionsError(str(e)) from e
+
+        response_code = request_json["response"].get("status", "")
+
+        response_content = request_json["response"]["content"].get("text", "")
+
+        response_headers = self.fix_headers(request_json["response"].get("headers", []))
+        try:
+            newflow.response = http.Response.make(
+                response_code, response_content, response_headers
+            )
+        except TypeError as e:
+            logger.error("Failed to create response")
+            raise exceptions.OptionsError(str(e)) from e
+
+        return newflow
+
+    async def load_flows(self) -> None:
+        """
+        Loads flows into mitmproxy
+
+        Raises:
+            exceptions.FlowReadException: If unable to load flow the FlowReadException is raised
+        """
+        cnt = 0
+        for flow in self.flows:
+            try:
+                await ctx.master.load_flow(flow)
+                cnt += 1
+            except (OSError, exceptions.FlowReadException) as e:
+                if cnt:
+                    logger.warning("Flow file corrupted - loaded %i flows." % cnt)
+                else:
+                    logger.error("Flow file corrupted.")
+                raise exceptions.FlowReadException(str(e)) from e
+
+    @command.command("readhar")
+    def read_har(
+        self,
+        path: types.Path,
+    ) -> None:
+        """
+        Reads a HAR file into mitmproxy. Loads a flow for each entry in given HAR file.
+
+        Args:
+            path (types.Path): Path to HAR file
+        """
+        with open(path) as fp:
+            try:
+                harFile = json.load(fp)
+            except Exception:
+                logger.error("HAR file was unable to be loaded using json.load()")
+                raise (
+                    Exception(
```

Comment:
If you raise a `mitmproxy.exceptions.CommandError`, this should look much nicer in the UI. We also don't need a separate log statement then. :)